### PR TITLE
Drop EPID

### PR DESCRIPTION
**Pull request by woju**
## Description of the changes 

See commit.

## How to test this PR? 

- CI.
- Read docs.

### CODE REVIEW

**Review by kailun-qin — CHANGES_REQUESTED** [review]
> Reviewed 67 of 73 files at r1, 7 of 7 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/2062)* status: all files reviewed, 12 unresolved discussions, not enough approvals from maintainers (1 more required), not enough approvals from different teams (1 more required, approved so far: ITL) (waiting on @woju)

___
*[`CI-Examples/ra-tls-secret-prov/Makefile` line 17 at r2](https://reviewable.io/reviews/gramineproject/gramine/2062#-OCYtLJB2i4UtManPUxL:-OCYtLJB2i4UtManPUxM:be0drob) ([raw file](https://github.com/gramineproject/gramine/blob/eda27b05bb1e6e651a20e53322ffc815617bc0ec/CI-Examples/ra-tls-secret-prov/Makefile#L17)):*
> ```Makefile
> LDFLAGS += -pie
> 
> RA_TYPE ?= dcap
> ```

Any special reason that we make this different from the `ra-tls-mbedtls` example where we remove `RA_TYPE` completely from the Makefile while hardcoding it in the manifest template?



_Code quote:_
```Makefile
RA_TYPE ?= dcap
```
___
*[`Documentation/attestation.rst` line 79 at r2](https://reviewable.io/reviews/gramineproject/gramine/2062#-OCYwzEh23xcpx9UQf3h:-OCYwzEh23xcpx9UQf3i:bxwkv8g) ([raw file](https://github.com/gramineproject/gramine/blob/eda27b05bb1e6e651a20e53322ffc815617bc0ec/Documentation/attestation.rst#L79)):*
> ```reStructuredText
> In turn, the Quoting Enclave uses the key provided by :term:`Provisioning
> Certification Enclave` (PCE; step 5, only during initial deployment of this SGX
> machine). The PCE uses an Intel service called :term:`Intel Provisioning
> ```

hmm.. Is this correct? What is "the key"? Isn['t the attestation key generated by the QE itself?



_Code quote:_
```reStructuredText
In turn, the Quoting Enclave uses the key provided by :term:`Provisioning
Certification Enclave` (PCE; step 5, only during initial deployment of this SGX
machine). The PCE uses an Intel service called :term:`Intel Provisioning
```
___
*[`CI-Examples/busybox/README.md` line 19 at r2](https://reviewable.io/reviews/gramineproject/gramine/2062#-OCYra3z8G7eBiHr3I1d:-OCYra3z8G7eBiHr3I1e:bbs0xpb) ([raw file](https://github.com/gramineproject/gramine/blob/eda27b05bb1e6e651a20e53322ffc815617bc0ec/CI-Examples/busybox/README.md#L19)):*
> ```Markdown
> If you want to try out [`/dev/attestation/` files][attestation], you must build
> the example with SGX remote attestation enabled. By default, the example is
> built *without* remote attestation.
> ```

This contradicts w/ the Makefile?



_Code quote:_
```Markdown
the example with SGX remote attestation enabled. By default, the example is
built *without* remote attestation.
```
___
*[`CI-Examples/python/README.md` line 25 at r2](https://reviewable.io/reviews/gramineproject/gramine/2062#-OCYrhH32nxxgcwSw4bL:-OCYrhH32nxxgcwSw4bM:b-7ropln) ([raw file](https://github.com/gramineproject/gramine/blob/eda27b05bb1e6e651a20e53322ffc815617bc0ec/CI-Examples/python/README.md#L25)):*
> ```Markdown
> If you want to run the `scripts/sgx-quote.py` script, you must build the example
> with SGX remote attestation enabled. By default, the example is built *without*
> remote attestation.
> ```

ditto



_Code quote:_
```Markdown
with SGX remote attestation enabled. By default, the example is built *without*
remote attestation.
```
___
*[`meson_options.txt` line 16 at r2](https://reviewable.io/reviews/gramineproject/gramine/2062#-OCYoh5C3lBUR2NLRSdI:-OCYoh5C3lBUR2NLRSdJ:bn9yxdw) ([raw file](https://github.com/gramineproject/gramine/blob/eda27b05bb1e6e651a20e53322ffc815617bc0ec/meson_options.txt#L16)):*
> ```Meson
> 
> option('dcap', type: 'feature',
>     description: 'Build additional utilities linked against DCAP library')
> ```

move this before the TODO above?



_Code quote:_
```Meson
option('dcap', type: 'feature',
    description: 'Build additional utilities linked against DCAP library')
```
___
*[`CI-Examples/python/Makefile` line 18 at r2](https://reviewable.io/reviews/gramineproject/gramine/2062#-OCYqF7t2e6F0a8PPa_J:-OCYqF7t2e6F0a8PPa_K:b-7ropln) ([raw file](https://github.com/gramineproject/gramine/blob/eda27b05bb1e6e651a20e53322ffc815617bc0ec/CI-Examples/python/Makefile#L18)):*
> ```Makefile
> endif
> 
> RA_TYPE ?= dcap
> ```

ditto



_Code quote:_
```Makefile
RA_TYPE ?= dcap
```
___
*[`Documentation/sgx-intro.rst` line 370 at r2](https://reviewable.io/reviews/gramineproject/gramine/2062#-OCZ1n-h4Hh2uPfavhl0:-OCZ1n-h4Hh2uPfavhl1:bdvw8ud) ([raw file](https://github.com/gramineproject/gramine/blob/eda27b05bb1e6e651a20e53322ffc815617bc0ec/Documentation/sgx-intro.rst#L370)):*
> ```reStructuredText
> 
>       The attestation mode being used can be chosen by the application enclave,
>       but it must match what was chosen when generating the :term:`SPID`.
> ```

What about here?



_Code quote:_
```reStructuredText
      The attestation mode being used can be chosen by the application enclave,
      but it must match what was chosen when generating the :term:`SPID`.
```
___
*[`Documentation/sgx-intro.rst` line 380 at r2](https://reviewable.io/reviews/gramineproject/gramine/2062#-OCZ1ps_2Vak2da0qjVV:-OCZ1ps_2Vak2da0qjVW:b-s5stkv) ([raw file](https://github.com/gramineproject/gramine/blob/eda27b05bb1e6e651a20e53322ffc815617bc0ec/Documentation/sgx-intro.rst#L380)):*
> ```reStructuredText
>          :term:`SPID`
>             An identifier one can obtain from Intel, required to make use of EPID
>             attestation.
> ```

What about here? We removed SPID almost everywhere.



_Code quote:_
```reStructuredText
         :term:`SPID`
            An identifier one can obtain from Intel, required to make use of EPID
            attestation.
```
___
*[`Documentation/sgx-intro.rst` line 736 at r2](https://reviewable.io/reviews/gramineproject/gramine/2062#-OCZ27Qg50VrjB8SEY7e:-OCZ27Qg50VrjB8SEY7f:bisdmcr) ([raw file](https://github.com/gramineproject/gramine/blob/eda27b05bb1e6e651a20e53322ffc815617bc0ec/Documentation/sgx-intro.rst#L736)):*
> ```reStructuredText
> 
>       See :term:`EPID` for a |~| description of the difference between
>       *linkable* and *unlinkable* quotes.
> ```

ditto, the link also indicates EOL only



_Code quote:_
```reStructuredText
   Service Provider ID
   SPID

      An identifier provided by Intel, used together with an |~| :term:`EPID`
      API key to authenticate to the :term:`Intel Attestation Service`. You can
      obtain an |~| SPID through Intel's `Trusted Services Portal
      <https://api.portal.trustedservices.intel.com/EPID-attestation>`_.

      See :term:`EPID` for a |~| description of the difference between
      *linkable* and *unlinkable* quotes.
```
___
*[`CI-Examples/busybox/Makefile` line 12 at r2](https://reviewable.io/reviews/gramineproject/gramine/2062#-OCYq5VRAoCTNyIt4Xld:-OCYq5VRAoCTNyIt4Xle:b7dyeki) ([raw file](https://github.com/gramineproject/gramine/blob/eda27b05bb1e6e651a20e53322ffc815617bc0ec/CI-Examples/busybox/Makefile#L12)):*
> ```Makefile
> endif
> 
> RA_TYPE ?= dcap
> ```

Why do we want to set dcap by default?



_Code quote:_
```Makefile
dcap
```
___
*[`tools/sgx/ra-tls/ra_tls.h` line 106 at r2](https://reviewable.io/reviews/gramineproject/gramine/2062#-OCZ3pae1C5R_We-QKRQ:-OCZ3pae1C5R_We-QKRR:brtnulh) ([raw file](https://github.com/gramineproject/gramine/blob/eda27b05bb1e6e651a20e53322ffc815617bc0ec/tools/sgx/ra-tls/ra_tls.h#L106)):*
> ```objc
>  * This function must be called from a non-mbedTLS verification callback, e.g., from a user-defined
>  * OpenSSL callback for SSL_CTX_set_cert_verify_callback(). All parameters required for the SGX
>  * quote, IAS attestation report verification, and/or DCAP quote verification must be passed in the
> ```

what about here?



_Code quote:_
```objc
IAS attestation report verification
```
___
*[`tools/sgx/ra-tls/ra_tls.h` line 124 at r2](https://reviewable.io/reviews/gramineproject/gramine/2062#-OCZ3w026VslAnqGUDuT:-OCZ3w026VslAnqGUDuU:b-7ropln) ([raw file](https://github.com/gramineproject/gramine/blob/eda27b05bb1e6e651a20e53322ffc815617bc0ec/tools/sgx/ra-tls/ra_tls.h#L124)):*
> ```objc
>  * This function must be called from a non-mbedTLS verification callback, e.g., from a user-defined
>  * OpenSSL callback for SSL_CTX_set_cert_verify_callback(). All parameters required for the SGX
>  * quote, IAS attestation report verification, and/or DCAP quote verification must be passed in the
> ```

ditto



_Code quote:_
```objc
IAS attestation report verification
```

**Review by mkow — CHANGES_REQUESTED** [review]
> Reviewed 6 of 73 files at r1, 2 of 7 files at r2, 1 of 9 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/2062)* status: 66 of 74 files reviewed, 12 unresolved discussions, not enough approvals from maintainers (1 more required), not enough approvals from different teams (1 more required, approved so far: ITL) (waiting on @kailun-qin)

___
*[`CI-Examples/busybox/Makefile` line 12 at r2](https://reviewable.io/reviews/gramineproject/gramine/2062#-OCYq5VRAoCTNyIt4Xld:-OCnq2En0iWor83Zpfvf:b-ew948i) ([raw file](https://github.com/gramineproject/gramine/blob/eda27b05bb1e6e651a20e53322ffc815617bc0ec/CI-Examples/busybox/Makefile#L12)):*
<details><summary><i>Previously, woju (Wojtek Porczyk) wrote…</i></summary><blockquote>

IMHO we don't have anything to gain by setting this to `none`. If you want to run in `gramine-direct`, it's ignored anyway. The only problem is when you run this in `gramine-sgx` on a machine with invalid provisioning, then you have bigger problems I believe.

This is in contrast to `python` and `ra-tls-*` examples, which simply do not work for `none` (see next discussion), where I kinda had to remove it. So right now I'm thinking we can get `RA_TLS ?= dcap` everywhere. Or should we even hardcode in manifest, and remove make variable? I'm not 100% sure at this point.
</blockquote></details>

I'd set it to `none` in examples that don't need it, we'd get some more settings coverage in tests, and it doesn't matter for this example.

**Review by woju — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/2062)* status: 65 of 74 files reviewed, 12 unresolved discussions, not enough approvals from maintainers (1 more required), not enough approvals from different teams (1 more required, approved so far: ITL) (waiting on @kailun-qin and @mkow)

___
*[`CI-Examples/busybox/Makefile` line 12 at r2](https://reviewable.io/reviews/gramineproject/gramine/2062#-OCYq5VRAoCTNyIt4Xld:-OD5g3Jd0_YyrjLKpSfl:b-706o2i) ([raw file](https://github.com/gramineproject/gramine/blob/eda27b05bb1e6e651a20e53322ffc815617bc0ec/CI-Examples/busybox/Makefile#L12)):*
<details><summary><i>Previously, mkow (Michał Kowalczyk) wrote…</i></summary><blockquote>

I'd set it to `none` in examples that don't need it, we'd get some more settings coverage in tests, and it doesn't matter for this example.
</blockquote></details>

Done. OK, that's fair.

**Review by kailun-qin — CHANGES_REQUESTED** [review]
> Reviewed 7 of 9 files at r3, 2 of 2 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/2062)* status: all files reviewed, 4 unresolved discussions, not enough approvals from maintainers (1 more required), not enough approvals from different teams (1 more required, approved so far: ITL) (waiting on @mkow and @woju)

___
*[`CI-Examples/python/Makefile` line 18 at r2](https://reviewable.io/reviews/gramineproject/gramine/2062#-OCYqF7t2e6F0a8PPa_J:-OD5r-fKAE7yTpsL37yD:b-x8tyw7) ([raw file](https://github.com/gramineproject/gramine/blob/eda27b05bb1e6e651a20e53322ffc815617bc0ec/CI-Examples/python/Makefile#L18)):*
<details><summary><i>Previously, woju (Wojtek Porczyk) wrote…</i></summary><blockquote>

`./run-tests.sh` checks if `SGX` is `1` and then runs `scripts/sgx-quote.py` and `scripts/sgx-report.py`. Those scripts directly operate on `/dev/attestation` interface, and it obviously fails if you set it to `none`. Again, either run `direct`, where this is ignored, or run with SGX and with `dcap`, no other configurations work.

This is even worse than `busybox` that just runs `echo Hello world`, which obviously works with non-functional `/dev/attestation`.
</blockquote></details>

Yeah, thanks for the explanation!
___
*[`Documentation/attestation.rst` line 79 at r2](https://reviewable.io/reviews/gramineproject/gramine/2062#-OCYwzEh23xcpx9UQf3h:-OD5uwQrAvSKeL-mvgly:b-awwbxg) ([raw file](https://github.com/gramineproject/gramine/blob/eda27b05bb1e6e651a20e53322ffc815617bc0ec/Documentation/attestation.rst#L79)):*
<details><summary><i>Previously, woju (Wojtek Porczyk) wrote…</i></summary><blockquote>

Yeah, this is wrong, was copied from EPID. How do I rewrite it?
</blockquote></details>

I would say sth like "The Quoting Enclave generates its own attestation key which is certified by PCE" (using the PCK, if you want to mention this also) .
___
*[`Documentation/sgx-intro.rst` line 370 at r2](https://reviewable.io/reviews/gramineproject/gramine/2062#-OCZ1n-h4Hh2uPfavhl0:-OD5s6eKDI6Glvc5Svus:bz12n7q) ([raw file](https://github.com/gramineproject/gramine/blob/eda27b05bb1e6e651a20e53322ffc815617bc0ec/Documentation/sgx-intro.rst#L370)):*
> because it's still true.

> Should I add some more disclaimers, like "EPID is deprecated \[...\] and support in Gramine was removed in version 1.9"?

Yes, it's still true. I'm fine w/ keeping it as is.
___
*[`Documentation/sgx-intro.rst` line 380 at r2](https://reviewable.io/reviews/gramineproject/gramine/2062#-OCZ1ps_2Vak2da0qjVV:-OD5rOE78hYROp44kbmc:b-mofdb2) ([raw file](https://github.com/gramineproject/gramine/blob/eda27b05bb1e6e651a20e53322ffc815617bc0ec/Documentation/sgx-intro.rst#L380)):*
> But it is still true, isn't it? I can reword it to past tense, something like "An identifier one could have obtained from Intel before EPID got EOL'd"?

Yes, it's still true. I'm fine w/ either rewording it to past tense or just leaving it as is.
___
*[`Documentation/attestation.rst` line 81 at r4](https://reviewable.io/reviews/gramineproject/gramine/2062#-OD62g4D5iEYEq6djNqu:-OD62g4D5iEYEq6djNqv:boypg7d) ([raw file](https://github.com/gramineproject/gramine/blob/5a44ddc9645f901d2c112a920f2b8e699aa48f42/Documentation/attestation.rst#L81)):*
> ```reStructuredText
> machine). The PCE uses an Intel service called :term:`Intel Provisioning
> Certification Service` (PCS) to obtain the attestation collateral: attestation
> certificates and certificate revocation lists for the SGX machine (step 6).
> ```

Well, it was originally written this way. But I don't think the PCE is responsible for obtaining attestation collateral and CRLs from PCS?



_Code quote:_
```reStructuredText
machine). The PCE uses an Intel service called :term:`Intel Provisioning
Certification Service` (PCS) to obtain the attestation collateral: attestation
certificates and certificate revocation lists for the SGX machine (step 6).
```

**Review by efu39 — COMMENTED** [review]
> Reviewed 64 of 73 files at r1, 5 of 7 files at r2, 8 of 9 files at r3, 2 of 2 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/2062)* status: all files reviewed, 6 unresolved discussions, not enough approvals from maintainers (1 more required), not enough approvals from different teams (1 more required, approved so far: ITL) (waiting on @mkow and @woju)

___
*[`CI-Examples/ra-tls-mbedtls/README.md` line 35 at r4](https://reviewable.io/reviews/gramineproject/gramine/2062#-OD7k9UD8chPmVg9PAqA:-OD7k9UD8chPmVg9PAqB:bmuy2zu) ([raw file](https://github.com/gramineproject/gramine/blob/5a44ddc9645f901d2c112a920f2b8e699aa48f42/CI-Examples/ra-tls-mbedtls/README.md#L35)):*
> ```Markdown
> The client is supposed to run on a trusted machine (*not* in an SGX enclave).
> If RA-TLS library `ra_tls_verify_dcap.so` is not requested by user via `dcap`
> command-line arguments respectively, the client falls back to using normal
> ```

argument,



_Code quote:_
```Markdown
arguments respectively,
```
___
*[`Documentation/tutorials/pytorch/index.rst` line 367 at r4](https://reviewable.io/reviews/gramineproject/gramine/2062#-OD7tX2j7ifr4auKduFY:-OD7tX2j7ifr4auKduFZ:bm1ybor) ([raw file](https://github.com/gramineproject/gramine/blob/5a44ddc9645f901d2c112a920f2b8e699aa48f42/Documentation/tutorials/pytorch/index.rst#L367)):*
> ```reStructuredText
>    git clone --depth 1 --branch v1.5 https://github.com/gramineproject/gramine.git
>    cd gramine/CI-Examples/ra-tls-secret-prov
>    make app dcap RA_TYPE=dcap
> ```

Should "RA_TYPE=dcap" be removed here?



_Code quote:_
```reStructuredText
RA_TYPE=dcap
```

**Review by woju — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/2062)* status: 70 of 74 files reviewed, 6 unresolved discussions, not enough approvals from maintainers (1 more required), not enough approvals from different teams (1 more required, approved so far: ITL) (waiting on @efu39, @kailun-qin, and @mkow)

___
*[`CI-Examples/ra-tls-mbedtls/README.md` line 35 at r4](https://reviewable.io/reviews/gramineproject/gramine/2062#-OD7k9UD8chPmVg9PAqA:-ODB9NnkAHY2CFmtACGr:b-896fix) ([raw file](https://github.com/gramineproject/gramine/blob/5a44ddc9645f901d2c112a920f2b8e699aa48f42/CI-Examples/ra-tls-mbedtls/README.md#L35)):*
<details><summary><i>Previously, efu39 (Erica Fu) wrote…</i></summary><blockquote>

argument,
</blockquote></details>

Done.
___
*[`Documentation/attestation.rst` line 79 at r2](https://reviewable.io/reviews/gramineproject/gramine/2062#-OCYwzEh23xcpx9UQf3h:-ODBATer5l7xbYBBGXZ3:b-896fix) ([raw file](https://github.com/gramineproject/gramine/blob/eda27b05bb1e6e651a20e53322ffc815617bc0ec/Documentation/attestation.rst#L79)):*
<details><summary><i>Previously, kailun-qin (Kailun Qin) wrote…</i></summary><blockquote>

I would say sth like "The Quoting Enclave generates its own attestation key which is certified by PCE" (using the PCK, if you want to mention this also) .
</blockquote></details>

Done.
___
*[`Documentation/attestation.rst` line 81 at r4](https://reviewable.io/reviews/gramineproject/gramine/2062#-OD62g4D5iEYEq6djNqu:-ODBAm1v34Sf6jVoqXBo:b-ccpfll) ([raw file](https://github.com/gramineproject/gramine/blob/5a44ddc9645f901d2c112a920f2b8e699aa48f42/Documentation/attestation.rst#L81)):*
<details><summary><i>Previously, kailun-qin (Kailun Qin) wrote…</i></summary><blockquote>

Well, it was originally written this way. But I don't think the PCE is responsible for obtaining attestation collateral and CRLs from PCS?
</blockquote></details>

Yeah, and the diagram is probably slightly wrong, because of the direction of the arrow.
___
*[`Documentation/sgx-intro.rst` line 380 at r2](https://reviewable.io/reviews/gramineproject/gramine/2062#-OCZ1ps_2Vak2da0qjVV:-ODBpyGpDik59bB0WvnA:b-1z9mvd) ([raw file](https://github.com/gramineproject/gramine/blob/eda27b05bb1e6e651a20e53322ffc815617bc0ec/Documentation/sgx-intro.rst#L380)):*
<details><summary><i>Previously, kailun-qin (Kailun Qin) wrote…</i></summary><blockquote>

> But it is still true, isn't it? I can reword it to past tense, something like "An identifier one could have obtained from Intel before EPID got EOL'd"?

Yes, it's still true. I'm fine w/ either rewording it to past tense or just leaving it as is.
</blockquote></details>

Reworded for past tense.
___
*[`Documentation/tutorials/pytorch/index.rst` line 367 at r4](https://reviewable.io/reviews/gramineproject/gramine/2062#-OD7tX2j7ifr4auKduFY:-ODBqDO9DfAQcBL4lku4:blp4x9v) ([raw file](https://github.com/gramineproject/gramine/blob/5a44ddc9645f901d2c112a920f2b8e699aa48f42/Documentation/tutorials/pytorch/index.rst#L367)):*
<details><summary><i>Previously, efu39 (Erica Fu) wrote…</i></summary><blockquote>

Should "RA_TYPE=dcap" be removed here?
</blockquote></details>

Done. Nice catch, thanks!

**Review by kailun-qin — CHANGES_REQUESTED** [review]
> Reviewed 1 of 9 files at r3, 4 of 4 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/2062)* status: all files reviewed, 5 unresolved discussions, not enough approvals from maintainers (1 more required), not enough approvals from different teams (1 more required, approved so far: ITL) (waiting on @efu39, @mkow, and @woju)

___
*[`Documentation/attestation.rst` line 81 at r4](https://reviewable.io/reviews/gramineproject/gramine/2062#-OD62g4D5iEYEq6djNqu:-ODEWSBZDaRit4euIY5w:b-61zbcr) ([raw file](https://github.com/gramineproject/gramine/blob/5a44ddc9645f901d2c112a920f2b8e699aa48f42/Documentation/attestation.rst#L81)):*
> `PCE` communicates with an Intel service called `Intel Provisioning Certification Service`

But PCE is an enclave that won't directly communicate with PCS (e.g., over networks) -- the issuing of PCK certificate is done thru registration (for multi-package) I think.

**Review by woju — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/2062)* status: all files reviewed, 4 unresolved discussions, not enough approvals from maintainers (1 more required), not enough approvals from different teams (1 more required, approved so far: ITL) (waiting on @efu39, @kailun-qin, and @mkow)

___
*[a discussion](https://reviewable.io/reviews/gramineproject/gramine/2062#-OD5zkB53VpuGU4D-P9j:-ODGiz5F3po4PnffgBHK:b-nkptwm) (no related file):*
<details><summary><i>Previously, woju (Wojtek Porczyk) wrote…</i></summary><blockquote>

CI failed on `ppoll01` (https://leeroy.cs.unc.edu/job/graphene-direct-24.04-sanitizers/163/console):
```
10:05:24  [1.428] /home/jenkins/workspace/graphene-direct-24.04-sanitizers/libos/test/ltp/src/testcases/kernel/syscalls/ppoll/ppoll01.c:246: TINFO: case SEND_SIGINT_RACE_TEST
10:05:24  [3.529] /home/jenkins/workspace/graphene-direct-24.04-sanitizers/libos/test/ltp/src/testcases/kernel/syscalls/ppoll/ppoll01.c:291: TFAIL: ret: 0, exp: -1, ret_errno: SUCCESS (0), exp_errno: EINTR (4)
```

IDK what to think about this one, I don't fully understand what's happening there.
</blockquote></details>

Looks like transient fault.

**Review by efu39 — COMMENTED** [review]
> Reviewed 2 of 4 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/2062)* status: all files reviewed, 2 unresolved discussions, not enough approvals from maintainers (1 more required), not enough approvals from different teams (1 more required, approved so far: ITL) (waiting on @kailun-qin and @mkow)

**Review by kailun-qin — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/2062)* status: all files reviewed, 1 unresolved discussion (waiting on @mkow)

___
*[`Documentation/attestation.rst` line 81 at r4](https://reviewable.io/reviews/gramineproject/gramine/2062#-OD62g4D5iEYEq6djNqu:-ODJMGB5CLzW7SbuRxHA:bfla123) ([raw file](https://github.com/gramineproject/gramine/blob/5a44ddc9645f901d2c112a920f2b8e699aa48f42/Documentation/attestation.rst#L81)):*
> I'm inclined not to delve into the details of PCKIDRetrievealTool and PCCS.

Gotcha, I'm not blocking.

> Just that arrow is misleading: right now it only represents trust relationship, but the transmission of data happens the other way around.

Can't we just update that too?

**Review by woju — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/2062)* status: 74 of 75 files reviewed, 1 unresolved discussion, not enough approvals from maintainers (1 more required), not enough approvals from different teams (1 more required, approved so far: ITL) (waiting on @mkow)

___
*[`Documentation/attestation.rst` line 81 at r4](https://reviewable.io/reviews/gramineproject/gramine/2062#-OD62g4D5iEYEq6djNqu:-ODLwJKC3ELqJZ7BClf2:b-z5fzx) ([raw file](https://github.com/gramineproject/gramine/blob/5a44ddc9645f901d2c112a920f2b8e699aa48f42/Documentation/attestation.rst#L81)):*
<details><summary><i>Previously, kailun-qin (Kailun Qin) wrote…</i></summary><blockquote>

> I'm inclined not to delve into the details of PCKIDRetrievealTool and PCCS.

Gotcha, I'm not blocking.

> Just that arrow is misleading: right now it only represents trust relationship, but the transmission of data happens the other way around.

Can't we just update that too?
</blockquote></details>

I've looked closer at the svg diagram. Best I can do is probably to remove the cloud that partially covered the arrow number 6 and leave just the arrow pointed both ways. I failed to edit this SVG further, Inkscape somehow produced wrong file, so I just removed the relevant `<path>`s from the file itself. For consistency I also removed cloud on arrow number 0 (both of those clouds explained nothing and only clouded *[sic]* the diagram). WDYT? (please see either rendered yourself, or at https://gramine--2062.org.readthedocs.build/en/2062/attestation.html)

**Review by mkow — CHANGES_REQUESTED** [review]
> Reviewed 50 of 73 files at r1, 4 of 7 files at r2, 6 of 9 files at r3, 2 of 2 files at r4, 4 of 4 files at r5, 1 of 1 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/2062)* status: all files reviewed, 2 unresolved discussions, not enough approvals from maintainers (1 more required), not enough approvals from different teams (1 more required, approved so far: ITL) (waiting on @woju)

___
*[`Documentation/sgx-intro.rst` line 122 at r6](https://reviewable.io/reviews/gramineproject/gramine/2062#-ODrhcPd0vONtNgm1ft5:-ODrhcPd0vONtNgm1ft6:undefined) ([raw file](https://github.com/gramineproject/gramine/blob/f8c782ae2fbde0054d94f9ebae1c8404cf8d7ea8/Documentation/sgx-intro.rst#L122)):*
> ```reStructuredText
> - https://github.com/intel/linux-sgx-driver -- older out-of-tree one, therefore
>   sometimes called "OOT". It's incompatible with kernels >= 5.11, so virtually
>   unuseable on supported distributions.
> ```





_Suggestion:_
```reStructuredText
unusable
```
___
*[`CI-Examples/ra-tls-mbedtls/README.md` line 18 at r6](https://reviewable.io/reviews/gramineproject/gramine/2062#-ODqkJbY0fH8uXTaKLim:-ODqkJbZ7CTn6Ztf92c-:bee188q) ([raw file](https://github.com/gramineproject/gramine/blob/f8c782ae2fbde0054d94f9ebae1c8404cf8d7ea8/CI-Examples/ra-tls-mbedtls/README.md#L18)):*
> ```Markdown
> DCAP software infrastructure must be installed and work correctly on the host.
> 
> more documentation about ECDSA (DCAP) remote attestation scheme, refer to
> ```

Missing "For" at the beginning?

**Review by kailun-qin — COMMENTED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/2062)* status: all files reviewed, 2 unresolved discussions, not enough approvals from maintainers (1 more required), not enough approvals from different teams (1 more required, approved so far: ITL) (waiting on @woju)

___
*[`Documentation/attestation.rst` line 81 at r4](https://reviewable.io/reviews/gramineproject/gramine/2062#-OD62g4D5iEYEq6djNqu:-ODsiQ1NC-5n4NmUZQi7:br7hh5v) ([raw file](https://github.com/gramineproject/gramine/blob/5a44ddc9645f901d2c112a920f2b8e699aa48f42/Documentation/attestation.rst#L81)):*
<details><summary><i>Previously, woju (Wojtek Porczyk) wrote…</i></summary><blockquote>

I've looked closer at the svg diagram. Best I can do is probably to remove the cloud that partially covered the arrow number 6 and leave just the arrow pointed both ways. I failed to edit this SVG further, Inkscape somehow produced wrong file, so I just removed the relevant `<path>`s from the file itself. For consistency I also removed cloud on arrow number 0 (both of those clouds explained nothing and only clouded *[sic]* the diagram). WDYT? (please see either rendered yourself, or at https://gramine--2062.org.readthedocs.build/en/2062/attestation.html)
</blockquote></details>

Thanks! I'm fine w/ it.

**Review by kailun-qin — APPROVED** [review]
> Reviewed 2 of 2 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/2062)* status: all files reviewed, 2 unresolved discussions (waiting on @woju)

**Review by woju — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/2062)* status: all files reviewed, 2 unresolved discussions (waiting on @mkow)

___
*[`CI-Examples/ra-tls-mbedtls/README.md` line 18 at r6](https://reviewable.io/reviews/gramineproject/gramine/2062#-ODqkJbY0fH8uXTaKLim:-ODuI8tsC3E1-1zjAywC:b-896fix) ([raw file](https://github.com/gramineproject/gramine/blob/f8c782ae2fbde0054d94f9ebae1c8404cf8d7ea8/CI-Examples/ra-tls-mbedtls/README.md#L18)):*
<details><summary><i>Previously, mkow (Michał Kowalczyk) wrote…</i></summary><blockquote>

Missing "For" at the beginning?
</blockquote></details>

Done.
___
*[`Documentation/sgx-intro.rst` line 122 at r6](https://reviewable.io/reviews/gramineproject/gramine/2062#-ODrhcPd0vONtNgm1ft5:-ODuJ8HkEwD4F1mv-int:bve6qsg) ([raw file](https://github.com/gramineproject/gramine/blob/f8c782ae2fbde0054d94f9ebae1c8404cf8d7ea8/Documentation/sgx-intro.rst#L122)):*
https://en.wiktionary.org/wiki/unuseable:
> unuseable (comparative more unuseable, superlative most unuseable)
> 1. Alternative spelling of unusable.

With quotes from XIX to XXI century. But I'll change, since it seems it fell out of Oxford Dictionary.

**Review by mkow — APPROVED** [review]
> Reviewed 2 of 2 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/2062)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved

**Review by mkow — APPROVED** [review]
> Reviewed 1 of 1 files at r8, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/2062)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved

**Review by kailun-qin — APPROVED** [review]
> Reviewed 1 of 1 files at r8, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/2062)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved

**Comment by woju** [comment]
Jenkins, test Jenkins-SGX-24.04-Sanitizers please
